crates/typst/src/text/font/color.rs

```diff
@@ -0,0 +1,321 @@
+    // embed the original SVG in another one that has the exact dimensions of
+    // the glyph, with a transform to make it fit. We also need to remove the
+    // viewBox, height and width attributes from the inner SVG, otherwise usvg
+    // takes into account these values to clip the embedded SVG.
+    make_svg_unsized(&mut data);
+    let wrapper_svg = format!(
+        r#"
+        <svg
+            width="{width}"
+            height="{height}"
+            viewBox="0 0 {width} {height}"
+            xmlns="http://www.w3.org/2000/svg">
+            <g transform="matrix(1 0 0 1 {tx} {ty})">
+            {inner}
+            </g>
+        </svg>
+    "#,
+        inner = data,
+        tx = -left,
+        ty = -top,
+    );
+
+    let image = Image::new(
+        wrapper_svg.as_bytes().into(),
+        typst::visualize::ImageFormat::Vector(typst::visualize::VectorFormat::Svg),
+        None,
+    )
+    .unwrap();
+    let position = Point::new(Abs::pt(left), Abs::pt(top) + upem);
+    let size = Axes::new(Abs::pt(width), Abs::pt(height));
+    frame.push(position, FrameItem::Image(image, size, Span::detached()));
+}
+
+/// Remove all size specifications (viewBox, width and height attributes) from a
+/// SVG document.
+fn make_svg_unsized(svg: &mut String) {
+    let mut viewbox_range = None;
+    let mut width_range = None;
+    let mut height_range = None;
+
+    let mut s = unscanny::Scanner::new(svg);
+
+    s.eat_until("<svg");
+    s.eat_if("<svg");
+    while !s.eat_if('>') {
+        s.eat_whitespace();
+        let start = s.cursor();
+        let attr_name = s.eat_until('=').trim();
+        // Eat the equal sign and the quote.
+        s.eat();
+        s.eat();
+        let mut escaped = false;
+        while escaped || !s.eat_if('"') {
+            escaped = s.eat() == Some('\\');
+        }
+        match attr_name {
+            "viewBox" => {
+                viewbox_range = Some(start..s.cursor());
+            }
+            "width" => {
+                width_range = Some(start..s.cursor());
+            }
+            "height" => {
+                height_range = Some(start..s.cursor());
+            }
+            _ => {}
+        }
+    }
+
+    /// Because we will remove some attributes, other ranges may need to be
+    /// shifted This function returns a mutable reference to a range (a) if it
+    /// should be shifted after another range (b) was deleted
+    fn should_shift<'a>(
+        a: &'a mut Option<std::ops::Range<usize>>,
+        b: &std::ops::Range<usize>,
+    ) -> Option<&'a mut std::ops::Range<usize>> {
+        // Is a after b?
+        let is_after = a.as_ref().map(|r| r.start > b.end).unwrap_or(false);
+        if is_after {
+            a.as_mut()
+        } else {
+            None
+        }
+    }
+
+    // Remove the `viewBox` attribute.
+    if let Some(range) = viewbox_range {
+        svg.replace_range(range.clone(), "");
```

Comment:
maybe you could also replace with whitespace of the same length to prevent unnecessary backshift and the need for `should_shift`.